src/components/OptionsSelector/BaseOptionsSelector.js

```diff
@@ -61,620 +53,652 @@ const defaultProps = {
+    }, []);
+
+    useEffect(() => {
+        // Unregister the shortcut before registering a new one to avoid lingering shortcut listener
+        enterSubscription.current();
+        if (!disableEnterShortCut) {
+            subscribeToEnterShortcut();
         }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [disableEnterShortCut]);
 
-        const itemIndex = option.index;
-        const sectionIndex = option.sectionIndex;
+    useEffect(() => {
+        if (props.isFocused) {
+            subscribeToEnterShortcut();
+            subscribeToCtrlEnterShortcut();
+        } else {
+            unSubscribeFromKeyboardShortcut();
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [props.isFocused]);
 
-        if (!lodashGet(this.state.sections, `[${sectionIndex}].data[${itemIndex}]`, null)) {
+    useEffect(() => {
+        const newSections = sliceSections();
+
+        if (prevPaginationPage.current !== paginationPage) {
+            prevPaginationPage.current = paginationPage;
+            setSections(newSections);
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [paginationPage]);
+
+    useEffect(() => {
+        setFocusedOption(allOptions[focusedIndex]);
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [focusedIndex]);
+
+    // eslint-disable-next-line rulesdir/prefer-early-return
+    useEffect(() => {
+        // Screen coming back into focus, for example
+        // when doing Cmd+Shift+K, then Cmd+K, then Cmd+Shift+K.
+        // Only applies to platforms that support keyboard shortcuts
+        if (isWebOrDesktop && isFocused && props.autoFocus && textInputRef.current) {
+            setTimeout(() => {
+                textInputRef.current.focus();
+            }, CONST.ANIMATED_TRANSITION);
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [isFocused, props.autoFocus]);
+
+    useEffect(() => {
+        const newSections = sliceSections();
+        const newOptions = flattenSections();
+
+        if (prevLocale.current !== props.preferredLocale) {
+            prevLocale.current = props.preferredLocale;
+            setAllOptions(newOptions);
+            setSections(newSections);
             return;
         }
 
-        this.list.scrollToLocation({sectionIndex, itemIndex, animated});
-    }
+        const newFocusedIndex = props.selectedOptions.length;
+        const prevFocusedOption = _.find(newOptions, (option) => focusedOption && option.keyForList === focusedOption.keyForList);
+        const prevFocusedOptionIndex = prevFocusedOption ? _.findIndex(newOptions, (option) => focusedOption && option.keyForList === focusedOption.keyForList) : undefined;
+
+        setSections(newSections);
+        setAllOptions(newOptions);
+        setFocusedIndex(prevFocusedOptionIndex || (_.isNumber(props.focusedIndex) ? props.focusedIndex : newFocusedIndex));
```

Comment:
Since this was moved into an effect on sections, it's run on the initial render and incorrectly sets the focused index to 0 when it should be 1. That's causing this regression https://github.com/Expensify/App/issues/37238.

More explanation [posted here](https://github.com/Expensify/App/issues/37238#issuecomment-1965599101).